setup.py

```diff
@@ -223,6 +223,39 @@ def run(self):
             sys.exit(128)
 
 
+class Plugin(SimpleCommand):
+    """Handle plugins"""
+
+    description = 'Handle plugins'
+    user_options = [
+        ("list", 'l', "List available plugins"),
+        ("install=", 'i', "Plugin to install")
+    ]
+
+    def initialize_options(self):
+        self.list = False  # pylint: disable=W0201
+        self.install = None  # pylint: disable=W0201
+
+    def run(self):
+
+        plugins_list = []
+        directory = os.path.join(BASE_PATH, "optional_plugins")
+        for plugin in list(Path(directory).glob("*/setup.py")):
+            plugins_list.append(plugin.parts[-2])
+
+        if self.list or (not self.list and not self.install):
+            print("List of available plugins:\n ", "\n  ".join(plugins_list))
+            return
+
+        if self.install in plugins_list:
+            action = ["install", "--user"]
```

Comment:
Some may prefer to install over a virtual env, where `--user` might not be needed. Maybe we could let users/developers to decide if --user should be used or not. What do you think?

Reply:
> Some may prefer to install over a virtual env, where `--user` might not be needed. Maybe we could let users/developers to decide if --user should be used or not. What do you think?

I'd expect people using a virtual-env to use the target 'install' instead of 'install-user' The idea of install-user is when you install directly in your local user.

Reply:
But,  then you should query if --user is available on `install` instead of `develop`, right?

Reply:
> But, then you should query if --user is available on `install` instead of `develop`, right?

Sorry, I don't get what you mean.  This new target is unrelated to `develop`, it's aimed to install avocado for testing/trying to people who don't necessarily knows a lot of about Python. And they will install it directly in their system or more likely in a test Virtual Machine.

Reply:
Sorry, ignore my last comment.

My question is: How someone using virtual env will install plugins? Sometimes system admins/users uses virtual env to isolate installs to avoid a wrong update will break some apps. IMO virtualenvs, are useful not only for development. Forcing users to have a VM just to try avocado is not the way to go.

Reply:
@beraldoleal I'm not forcing anyone to have a VM, I'm trying to add a new option to cater a type of users I feel are common. 
This doesn't change things for users of virtual envs, they can continue using the target `install` and in fact, they get a new option because they can use `python setup.py plugin` to install plugins!

Reply:
> This doesn't change things for users of virtual envs, they can continue using the target install and in fact, they get a new option because they can use python setup.py plugin to install plugins!

This is not what I'm getting. Back to my original comment: this new option (I'm talking about the setup.py Plugin command) is not working for virtual env users.

When I type `python3 setup.py --help-commands  | grep plugin`, I will assume this will work, but is not:
```
(avocado-py3.7) [local@laptop avocado]$ python3 setup.py --help-commands | grep plugin
  plugin            Handle plugins
(avocado-py3.7) [local@laptop avocado]$ python3 setup.py plugin --install=html
...

Please make the appropriate changes for your system and try again.
Traceback (most recent call last):
  File "setup.py", line 394, in <module>
    install_requires=['setuptools'])
  File "/home/local/src/venvs/avocado-py3.7/lib64/python3.7/site-packages/setuptools/__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib64/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib64/python3.7/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/usr/lib64/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "setup.py", line 253, in run
    run([sys.executable, "setup.py"] + action, cwd=parent_dir, check=True)
  File "/usr/lib64/python3.7/subprocess.py", line 512, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['/home/local/src/venvs/avocado-py3.7/bin/python3', 'setup.py', 'install', '--user']' returned non-zero exit status 1.
```

As a Python contributor, the first thing that I will look is for the `setup.py`, ignoring the Makefile. I understand that some users will look first to Makefile targets, ignoring setup.py, but unless I'm missing something, we could do things a little bit more predictable/consistent here.

Reply:
@beraldoleal Could you please share all the steps? It works here with a small caveat: I have to setup PYTHONPATH, which I agree is not great, but I don't get that kind of error messages

Reply:
I tried on a fresh environment Fedora 33:

```
$ python3.7 -m venv venv
$ source venv/bin/activate
$ git clone -b plugin_update_makefile https://github.com/ana/avocado.git
$ cd avocado/
$  python3 setup.py plugin --install=html
```
And If I remove `--user` here, it works fine.

I you can't reproduce, I can try to investigate better, the real problem here.

Reply:
@beraldoleal Thanks! This is what I get https://gist.github.com/ana/c07ebceeedaba5ebb8f65d9351d74f87
Don't bother trying to reproduce yet, let's first sync the changes made to setup.py/Makefile in the last weeks and how we could finish this epic.